Allure steps for s3-tests. Fixes #39

MaxGelbakhiani opened the pull request:
Followed @evgeniiz321 review advice and implemented client methods overload with allure wraps.

roman-khimov commented on s3tests_boto3/functional/__init__.py:
```diff
@@ -14,18 +15,23 @@
 
+boto3._get_default_session = lambda: WrappedSession()
 
+@allure.step
```
Nameless? What's it gonna look like in the log?

roman-khimov commented:
What's up with this? No updates in two days.

MaxGelbakhiani commented:
> What's up with this? No updates in two days.

Added names to allure steps and decorator for functions getting settings from config. Also removed @allure.step from some functions where necessary in compare to the original commit.

roman-khimov commented on s3tests_boto3/functional/test_s3_neofs.py:
```diff
@@ -27,6 +28,8 @@
 
 
+
+@allure.step
```
Left intentionally (and others in the same file)?

MaxGelbakhiani replied:
Fixed

roman-khimov commented on s3tests_boto3/functional/utils.py:
```diff
@@ -1,7 +1,9 @@
 import string
 
 
+@allure.step("Assert Raises")
```
Is it a proper step, btw?

MaxGelbakhiani replied:
In my opinion, it is. As for tests, checking negative scenarios assertion with expected error is just a step. Like in [test_object_copy_bucket_not_found](https://github.com/nspcc-dev/s3-tests/blob/a35437900562ae3b17797fe82ca7c2c1840665fe/s3tests_boto3/functional/test_s3.py#L6768C56-L6768C56)

And it looks like this in test report:
<img width="652" alt="image" src="https://github.com/nspcc-dev/s3-tests/assets/13661886/45a3d3ff-93ab-45e5-8c7d-6efdebedfbd0">
